src/Microsoft.ML.Vision/ImageClassificationTrainer.cs

```diff
@@ -519,6 +527,17 @@ internal ImageClassificationTrainer(IHostEnvironment env, Options options)
             {
                 options.WorkspacePath = GetTemporaryDirectory();
             }
+            else
+            {
+                //Since this is not a tmp workspace, but one provided by the user, do not allow deletion.
+                options.CleanTmpWorkspace = false;
```

Comment:
add a readonly private variable and set it to true indicating we need to cleanup workspace in the event user does not provide one.